### PR TITLE
Add workers to Octo

### PR DESCRIPTION
**Pull request by MichaelJCompton**
Patterned off similar endpoints for Environments and Machines.

### CODE REVIEW

File: source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            options.Add("calamari-outdated=", "[Optional] State of Calamari to clean up. By default ignores Calamari state.", v => SetFlagState(v, ref isCalamariOutdated));
+            options.Add("tentacle-outdated=", "[Optional] State of Tentacle version to clean up. By default ignores Tentacle state", v => SetFlagState(v, ref isTentacleOutdated));
+        }
+
+        public async Task Request()
+        {
+            if (string.IsNullOrWhiteSpace(poolName))
+                throw new CommandException("Please specify a worker pool name using the parameter: --workerpool=XYZ");
+            if (!healthStatuses.Any() && !statuses.Any())
+                throw new CommandException("Please specify a status using the parameter: --health-status");
```
Comment (alfhenrik) [inline source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs]: Should this mention the `status` parameter as you can use either or both the `health-status` and `status` parameters to filter the machines.

File: source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+            workerpoolResource = await GetWorkerPool().ConfigureAwait(false);
```
Comment (alfhenrik) [inline source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs]: Such nitpick but OCD 😇 should be `workerPoolResource`

File: source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs (hunk 1 above)
Comment (MichaelJCompton) [reply source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs]: I interpreted this https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Cli/Commands/HealthStatusProvider.cs#L13-L16 as meaning that the `status` was for backward compatibility.

I thought cause worker pools is new, it doesn't need the backward compatibility ??

Comment (MichaelJCompton) [reply source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs]: ooohh, whoops, if I meant what I said I would have removed the status option.

Comment (MichaelJCompton) [reply source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs]: ok, I'm just confused, will redo.

**Pull request merged**